libexec/jenv-with

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+#
+# Summary: Execute command with specific Java version
+#
+# Usage: jenv with <version> <command...>
+#
+# Executes the command with specified Java version being effective.
+# This is equivalent to running `jenv shell <version>' followed by
+# the command, but without changing the shell environment.
+#
+# <version> should be a string matching a Java version known to jenv.
+# The special version string `system' will use your default system Java.
+# Run `jenv versions' for a list of available Java versions.
+
+set -e
+[ -n "$JENV_DEBUG" ] && set -x
+
+if [ "$#" -lt 2 ]; then
+  jenv-help --usage with >&2
+  exit 1
+fi
+
+# Provide jenv completions
+if [ "$1" = "--complete" ]; then
+  echo --unset
+  echo system
+  exec jenv-versions --bare
+fi
+
+export JENV_VERSION="$1"
+shift
+
+# Make the version effective (same as _jenv_export_hook)
+export JAVA_HOME
+JAVA_HOME=$(jenv-javahome)
```

Comment:
Thought on this @findepi ?
> It might be good to tweak `jenv-with` so that `jenv with system ...` also works?
>
> ```
> $ jenv with system jshell; echo $?
> Using system JDK, no JAVA_HOME set!
> 1
> ```

I _think_ it would be enough to make these
```sh
export JAVA_HOME=""
[ "system" = "$JENV_VERSION" ] && JAVA_HOME=$(jenv-javahome)
```

We _should_ also reset `JDK_HOME=""` here so if `javac` doesn't exist in `$JAVA_HOME` it doesn't preserve it's last value.

This is actually a bug in the exports hooks as well but I can push something separate for that.

Reply:
Done!